### PR TITLE
Show a message if no snaps are returned on store

### PR DESCRIPTION
**Pull request by Lukewh**
Fixes https://github.com/canonical-web-and-design/snapcraft.io/issues/2334

## QA
- Pull the branch
- Add `featured_snaps = []` to `webapp/store/views.py:79`
- `./run`
- Visit http://0.0.0.0:8004/store - you should receive an error page

Screenshot_2019-11-04 Maintenance mode enabled

### CODE REVIEW

**Comment by bartaz** [comment]
I wonder if we shouldn't just show an error page in such case.
Anything we show on this page won't work anyway probably - neither the search nor the 'See more'

Also, pretending it's a normal page makes the error message less 'visible', because the page just looks empty/broken until you read the text. It doesn't look like an error page.

Screenshot 2019-10-29 at 12 39 31

**Comment by clagom** [comment]
If neither the search nor the 'see more...' would work then yes I would say an error page makes sense @bartaz. Can we find out that those won't work? If users can still do those actions then I think the solution in the screenshot would work, otherwise a more global error without further call to actions would do.

**Comment by bartaz** [comment]
@clagom 

Because it's an error state from API it's really hard to say what will work and what wont.
It may have been just a temporary issue, so even refreshing the page may work or search or 'See more'. But if API is down probably neither of those will, but we wont know until user goes to these pages.

@Lukewh how is this state possible? API didn't return an error, but list of snaps was empty?

**Comment by Lukewh** [comment]
@bartaz exactly. Perhaps a better solution would be not show the featured section at all, just have the search box?

**Comment by clagom** [comment]
We have two examples here:
Screenshot 2019-10-29 at 13 30 23

Screenshot 2019-10-29 at 13 30 16

I would double check if we could at least still display the search bar

**Comment by Lukewh** [comment]
This is now updated to simply 503 when featured snaps are not returned.

File: webapp/store/views.py [hunk 1]
```diff
@@ -62,18 +62,19 @@ def store_view():
...
 
         categories = logic.get_categories(categories_results)
 
         try:
             featured_snaps_results = api.get_searched_snaps(
                 snap_searched="", category="featured", size=10, page=1
             )
-        except ApiError:
+        except ApiError as api_error:
             featured_snaps_results = []
```
Comment (bartaz) [inline webapp/store/views.py]: Probably not needed anymore if we just `flask.abort` couple lines later?

File: webapp/store/views.py [hunk 2]
```diff
@@ -62,18 +62,19 @@ def store_view():
...
+            status_code, error_info = _handle_errors(api_error)
+            flask.abort(status_code)
```
Comment (bartaz) [inline webapp/store/views.py]: Should it be `return flask.abort` (at least it seems what we are doing elsewhere).
```suggestion
            return flask.abort(status_code)
```

File: webapp/store/views.py (hunk 1 above)
Comment (Lukewh) [reply webapp/store/views.py]: The API returning non-200 and there being no snaps are two different cases. I figure we should show an error case for both, so the second check is for the latter case.

**Pull request merged**